From 27bd55b86443e38e8d6c2d6b78fd21d4e3abdeef Mon Sep 17 00:00:00 2001
From: David Davis <daviddavis@users.noreply.github.com>
Date: Thu, 12 Jan 2023 08:55:17 -0500
Subject: [PATCH] Exclude pulp_deb's ReleaseFile from orphan cleanup

This is a temporary workaround to let orphan cleanup proceed until the
follow pulp_deb issue is fixed.

https://github.com/pulp/pulp_deb/issues/690
---
 pulpcore/app/tasks/orphan.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pulpcore/app/tasks/orphan.py b/pulpcore/app/tasks/orphan.py
index 471ed4f08..37d126bb5 100644
--- a/pulpcore/app/tasks/orphan.py
+++ b/pulpcore/app/tasks/orphan.py
@@ -51,7 +51,7 @@ def orphan_cleanup(content_pks=None, orphan_protection_time=settings.ORPHAN_PROT
 
     while True:
         content = Content.objects.orphaned(orphan_protection_time, content_pks).exclude(
-            pulp_type=PublishedMetadata.get_pulp_type()
+            pulp_type__in=[PublishedMetadata.get_pulp_type(), "deb.release_file"]
         )
         content_count = content.count()
         if not content_count:
-- 
2.34.1

